src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java

```diff
@@ -88,15 +95,46 @@ public CallbackGetRequestModel preProcess(Request request, Response response)
         @Nullable Object accessToken = tokenResponseData.get("access_token");
         Map<String, String> userInfoResponseData = getUserInfo(accessToken);
         List<Attribute> subjectAttributes = new LinkedList<>(), contextAttributes = new LinkedList<>();
-
-        // TODO: Build attribute collections
-
-        // FIXME: Org check
-        //checkUserOrganizationMembership(authenticationResult, tokenMap.get(PARAM_ACCESS_TOKEN).toString());
-
-        // TODO: Send back actual authentication result
-
-        return Optional.empty();
+        String login = userInfoResponseData.get("login");
+
+        subjectAttributes.add(Attribute.of("subject",  login));
+        subjectAttributes.add(Attribute.of("name", Name.of(userInfoResponseData.get("name"))));
+        subjectAttributes.add(Attribute.of("photo", Photo.of(userInfoResponseData.get("avatar_url"), false)));
+        subjectAttributes.add(Attribute.of("repos_url", userInfoResponseData.get("repos_url")));
+        subjectAttributes.add(Attribute.of("gists_url", userInfoResponseData.get("gists_url")));
+        subjectAttributes.add(Attribute.of("following_url", userInfoResponseData.get("following_url")));
+        subjectAttributes.add(Attribute.of("bio", userInfoResponseData.get("bio")));
+        subjectAttributes.add(Attribute.of("starred_url", userInfoResponseData.get("starred_url")));
+        subjectAttributes.add(Attribute.of("blog", userInfoResponseData.get("blog")));
+        subjectAttributes.add(Attribute.of("url", userInfoResponseData.get("url")));
+        subjectAttributes.add(Attribute.of("subscriptions_url", userInfoResponseData.get("subscriptions_url")));
+        subjectAttributes.add(Attribute.of("received_events_url", userInfoResponseData.get("received_events_url")));
+        subjectAttributes.add(Attribute.of("events_url", userInfoResponseData.get("events_url")));
+        subjectAttributes.add(Attribute.of("html_url", userInfoResponseData.get("html_url")));
+        subjectAttributes.add(Attribute.of("location", userInfoResponseData.get("location")));
+        subjectAttributes.add(Attribute.of("company", userInfoResponseData.get("company")));
+        subjectAttributes.add(Attribute.of("gravatar_id", userInfoResponseData.get("gravatar_id")));
+        subjectAttributes.add(Attribute.of("organizations_url", userInfoResponseData.get("organizations_url")));
+        _config.getManageOrganization().ifPresent(manageOrganization -> {
+            manageOrganization.getOrganizationName().ifPresent(organizationName -> {
+                subjectAttributes.add(Attribute.of("organization_name", organizationName));
```

Comment:
why isn't this set all the time it's present?

Reply:
`Manage Organization` is optional, so organization name will only exist if the manage organization is enabled